### PR TITLE
Add frontend attributes to HLO module

### PR DESCRIPTION
**Pull request by sogartar**
This is a PR that addresses #5424.

### CODE REVIEW

**Comment by sogartar** [comment]
It seems that when I formatted with `clang-format` a bunch of files got reformatted. Is this supposed to happen?

**Comment by radhakrishnaba** [comment]
@burmako Any update on this?

File: xla/hlo/ir/hlo_frontend_attributes.cc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (sergachev) [inline xla/hlo/ir/hlo_frontend_attributes.cc:1]: 2023

File: xla/hlo/ir/hlo_frontend_attributes.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (sergachev) [inline xla/hlo/ir/hlo_frontend_attributes.h:1]: 2023

**Review by sergachev — APPROVED** [review]

File: xla/hlo/ir/hlo_frontend_attributes.cc (hunk 1 above)
Comment (sogartar) [reply xla/hlo/ir/hlo_frontend_attributes.cc]: Done.

File: xla/hlo/ir/hlo_frontend_attributes.h (hunk 2 above)
Comment (sogartar) [reply xla/hlo/ir/hlo_frontend_attributes.h]: Done.

**Review by sergachev — APPROVED** [review]

**Pull request closed without merging**